frontend/src/scenes/events/EventsTable.js

```diff
@@ -1,281 +1,106 @@
-                    )}
-                    {event.elements.length > 0 &&
-                        event.elements[0].text &&
-                        ' with text "' + event.elements[0].text + '"'}
-                </td>
-                <td>
-                    <Link to={'/person/' + encodeURIComponent(event.distinct_id)} className="ph-no-capture">
-                        {event.person}
-                    </Link>
-                </td>
-                {params.map(paramRequest => {
-                    let param = paramRequest
-                    let value = event.properties[param]
+export function EventsTable({ fixedFilters, filtersEnabled = true }) {
+    const logic = eventsTableLogic({ fixedFilters })
+    const {
+        properties,
+        events,
+        isLoading,
+        hasNext,
+        isLoadingNext,
+        selectedEvent,
+        newEvents,
+        highlightEvents,
+    } = useValues(logic)
+    const { updateProperty, setSelectedEvent, fetchNextEvents, flipSort, prependNewEvents } = useActions(logic)
+    const {
+        location: { search },
+    } = useValues(router)
 
-                    if (param === '$current_url' && !value) {
-                        param = '$screen'
-                        value = event.properties[param]
-                    }
+    const showLinkToPerson = !fixedFilters?.person_id
 
-                    return (
-                        <td key={param} title={value}>
-                            <FilterLink
-                                property={param}
-                                value={event.properties[param]}
-                                filters={properties}
-                                onClick={(key, value) =>
-                                    this.setState(
-                                        {
-                                            properties: {
-                                                ...properties,
-                                                [key]: value,
-                                            },
-                                        },
-                                        this.fetchEvents
-                                    )
-                                }
-                            />
+    return (
+        <div className="events">
+            {filtersEnabled ? <PropertyFilters pageKey="EventsTable" /> : null}
+            <table className="table" style={{ position: 'relative' }}>
+                <thead>
+                    <tr>
+                        <th>Event</th>
+                        <th>Person</th>
+                        <th>Path / Screen</th>
+                        <th>Source</th>
+                        <th onClick={flipSort}>
+                            When <i className="fi flaticon-sort" />
+                        </th>
+                    </tr>
+                </thead>
+                <tbody>
+                    {isLoading && <TableRowLoading colSpan={5} />}
```

Comment:
Good catch! I didn't test changing filters in this table after changing the loaders, even though it's quite an obvious issue. I fixed it now for this table and for the others: if there are any rows already shown in the table, the loader is an overlay. Otherwise it's like on your screenshot, except of course without the table rows.

For added joy, I implemented a feature for the events table, which will show the loading overlay only after it has been loading for 500ms. Thus it's possible most people will never even see a loader anymore, except on the initial page load.

I can change that delay in case you have some better number to suggest.

Reply:
Looks good